src/open_dread_rando/door_locks/custom_door_types.py

```diff
@@ -375,34 +374,29 @@ def patch_material_data(self, editor: PatcherEditor, version: str):
             for mat_dat in self.data.alternate_mats:
                 create_custom_material(editor, mat_dat)
 
-    def add_weakness(self, weakness: str, new_template: dict):
-        life_funcs: list = new_template["property"]["components"]["LIFE"]["functions"]
+    def add_weakness(self, weakness: str, new_template: Bmsad) -> None:
+        life = new_template.components["LIFE"]
 
-        new_func = {
-            "name": "AddDamageSource",
-            "unk": 1,
-            "params": {
-                "Param1": {
-                    "type": "s",
-                    "value": weakness
-                }
-            }
-        }
+        new_func = ActorDefFunc.new("AddDamageSource")
+        new_func.set_param(1, weakness)
 
+        life_funcs = list(life.functions)
         life_funcs.append(new_func)
+        life.functions = life_funcs
```

Comment:
we could do `life.functions += (new_func,)` technically, I think. up to you if you think that's better or worse